assets/main/app.js

```diff
@@ -0,0 +1,105 @@
+String.prototype.hasVowels = function() {
+  var vowels = /[aeiou]/i;
+  return vowels.test(this);
+};
+
+String.prototype.toUpper = function() {
+  var upperCased = '';
+  for (var index = 0; index < this.length; index++) {
+    var currentValue = this.charCodeAt(index);
+    if (currentValue > 96 && currentValue < 123) {
+      upperCased += String.fromCharCode(currentValue & ~(1 << 5));
```

Comment:
You may need to add a comment on this line and also evaluate if you need to favour a one-line implementation over code readability.